### PR TITLE
feat(DataTable): adding support for dynamic rowDataHook

### PR DESCRIPTION
**Pull request by selama**
## Before you request a review, please make sure to check the following boxes:

- [ ] Explain what has changed.
- [ ] Explain why the change is needed.
- [ ] Added tests.
- [ ] Update documentation.
- [ ] Merge master with no conflicts.
- [ ] Make sure the build is passing on ci.
- [ ] If this pr closes any issue, please add a github link to close it.
- [ ] If you changed something in the product design/UX, you got the approval of [Ben Benhorin](https://wix.slack.com/messages/@benb).

### CODE REVIEW

**Comment by selama** [comment]
added support for dynamic rowDataHook.
it simplifies the ability to check whether the DataTable is rendered with correct data.

File: src/DataTable/DataTable.driver.js [hunk 1]
```diff
@@ -17,6 +17,8 @@ const dataTableDriverFactory = ({element, wrapper, component}) => {
   return {
     getRowsCount,
     getRowsWithClassCount: className => values(getRows()).filter(elem => elem.classList.contains(className)).length,
+    getRowsWithDataHookCount: dataHookName => element.querySelectorAll(`[data-hook="${dataHookName}"]`).length,
```
Comment (lbelinsk) [inline src/DataTable/DataTable.driver.js]: Code duplication here with the function
```
dataHookName => element.querySelectorAll(`[data-hook="${dataHookName}"]`)
```

Please extract it to a separated internal func at the top and use it.

File: src/DataTable/DataTable.spec.js [hunk 2]
```diff
@@ -76,6 +76,21 @@ describe('Table', () => {
     expect(driver.getRowsWithClassCount(defaultProps.rowClass)).toBe(defaultProps.data.length);
   });
 
+  it('should assign dataHook to rows', () => {
+    const rowDataHook = 'row-data-hook';
+    const props = Object.assign({}, defaultProps, {rowDataHook});
+    const driver = createDriver(<DataTable {...props}/>);
+    expect(driver.getRowsWithDataHookCount(rowDataHook)).toBe(defaultProps.data.length);
```
Comment (lbelinsk) [inline src/DataTable/DataTable.spec.js]: Please add another expect to check that the dataHook is correct. I mean add something similar to the next tets you wrote (line 91 for example)

File: src/DataTable/DataTable.js [hunk 3]
```diff
@@ -80,8 +79,7 @@ class DataTable extends WixComponent {
         loader={this.props.loader}
         useWindow={this.props.useWindow}
         scrollElement={this.props.scrollElement}
-        >
-        {table}
+        >{table}
```
Comment (lbelinsk) [inline src/DataTable/DataTable.js]: why?

File: src/DataTable/DataTable.driver.js (hunk 1 above)
Comment (selama) [reply src/DataTable/DataTable.driver.js]: it isn't exactly a duplication: in getRowsWithDataHookCount querySelectorAll is used, and in getRowWithDataHook querySelector is used.

I could used querySelectorAll for getRowWithDataHook as well(and pop the first result), but I'm not sure it is good practice.

what do you say?

File: src/DataTable/DataTable.spec.js (hunk 2 above)
Comment (selama) [reply src/DataTable/DataTable.spec.js]: on it.

File: src/DataTable/DataTable.js (hunk 3 above)
Comment (selama) [reply src/DataTable/DataTable.js]: I was strangling the es linter - changing it back to the original code style.

File: src/DataTable/DataTable.driver.js (hunk 1 above)
Comment (lbelinsk) [reply src/DataTable/DataTable.driver.js]: Right.. missed that. just leave it like that then...

**Pull request merged**